features/before_after_all_hooks_parallel.feature

```diff
@@ -0,0 +1,54 @@
+Feature: Before/AfterAll hooks in parallel
+
+  When running in parallel, Cucumber-JS runs some code on a "Co-ordinator" process,
+  but all the user code runs in a "Worker" process.
+
+  When defining BeforeAll/AfterAll hooks, you can choose whather to run them on
+  the coordinator, the worker, or both.
+
+  Background:
+    Given a file named "features/a.feature" with:
+      """
+      Feature: some feature
+        Scenario: first scenario
+          Given first step
+
+        Scenario: second scenario
+          Given second step
+      """
+
+  Scenario: Just run the AfterAll hooks once, on the coordinator
+
+    Given a file named "features/support/hooks.js" with:
+      """
+      const {AfterAll, HookParallelMode, Given} = require('@cucumber/cucumber')
+      const {expect} = require('chai')
+
+      let counter = 0
+
+      Given('first step', function() {
+        counter ++
+        expect(counter).to.eql(1)
+      })
+
+      Given('second step', function() {
+        counter ++
+        expect(counter).to.eql(1)
+      })
+
+      AfterAll({ parallelMode: HookParallelMode.WORKER_ONLY }, function() {
+        counter ++
+        console.log('worker: ', counter)
+        expect(counter).to.eql(2)
```

Comment:
I'm not sure this will work once we implement the changes - because the workers are in different Node.js processes, they won't be acting on the same `counter` variable, so I think it will always be `1` assuming the workers take one scenario each.

What might be worth pursuing instead is having each flavour of the hook `console.log` a string with the some of environment variables that Cucumber sets for the parallel runtime e.g `CUCUMBER_WORKER_ID` and then assert on that being in the output with a `Then the output contains the text:` step (see other features) - a bit messy but should work.